### PR TITLE
Form - Dynamic Text Solution

### PR DESCRIPTION
**Pull request by ifindpokemons**
e.g. Unown - B

## Description
This uses the 'form' attribute in the webhook supplied by RocketMap PR 1914 to display name - form e.g. Unown - B

## How Has This Been Tested?
My scan servers using RocketMap with 1914 applied. May not work with RocketMap not patched with PR 1914 someone should test that.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by ifindpokemons** [comment]
As requested, putting this in the v3.2 branch. Also looks like the hk locale stuff you added into the master branch has come across as well. Bonus!

File: locales/de/forms.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (kvangent) [inline locales/de/forms.json]: These should really be PkmnID => Form# => FormName.

This way when additional forms are added they can be added per pokemon (or in this case pokemon id).

File: PokeAlarm/WebhookStructs.py [hunk 2]
```diff
@@ -71,7 +71,8 @@ def pokemon(data):
             'gender': get_pokemon_gender(check_for_none(int, data.get('gender'), '?')),
             'size': 'unknown',
             'gmaps': get_gmaps_link(lat, lng),
-            'applemaps': get_applemaps_link(lat, lng)
+            'applemaps': get_applemaps_link(lat, lng),
+	        'form': check_for_none(int, data.get('form'), 0)
```
Comment (kvangent) [inline PokeAlarm/WebhookStructs.py]: Lets move 'form' up above gmaps (just for better logical grouping)

File: filters.json.example [hunk 3]
```diff
@@ -14,7 +14,7 @@
         "filters":{ "min_dist":"0", "max_dist":"10" }
     },
     "pokemon":{
-        "enabled":"False",
+        "enabled":"True",
```
Comment (kvangent) [inline filters.json.example]: This file doesn't need to be changed.

**Comment by kvangent** [comment]
I think it may be best that `<form>` sends an empty string for pokemon that aren't missing info but don't have a form set.

File: filters.json.example (hunk 3 above)
Comment (ifindpokemons) [inline filters.json.example]: Okies I'll revert this change.

File: locales/de/forms.json (hunk 1 above)
Comment (ifindpokemons) [inline locales/de/forms.json]: Yip, done :D json file structured to deliver array of forms per id, in teh code I build a 2d array [pkmn_id, form_int] of all forms available per id, so if event happens some pokemon come with form we can just alter json locale file and we are good to go.

**Comment by ifindpokemons** [comment]
@kvangent all done, form will return '' when there is not a form as requested. json updated to provide array of forms per pokemon id.
reverted the filters.json.example and also did the cosmetic movement of form above map links. :)

**Comment by imp3tus** [comment]
This PR has really evolved into a really well flowing code for unown form. Looking forward to this being in the master branch.

File: PokeAlarm/Manager.py [hunk 4]
```diff
@@ -482,6 +483,15 @@ def process_pokemon(self, pkmn):
             log.info("{} rejected: not inside geofence(s)".format(name))
             return
 
+        # Get the form character however because not all Pokemon have a form
+        # we must catch the exception when no form for pkmn_id in 2d array
+        # form character will remain null if not exists as requested
```
Comment (ifindpokemons) [inline PokeAlarm/Manager.py]: Opps I thing I meant empty not null :/

File: PokeAlarm/WebhookStructs.py [hunk 5]
```diff
@@ -70,6 +70,7 @@ def pokemon(data):
             'weight': check_for_none(float, data.get('weight'), 'unkn'),
             'gender': get_pokemon_gender(check_for_none(int, data.get('gender'), '?')),
             'size': 'unknown',
+            'form': check_for_none(int, data.get('form'), 0),
```
Comment (kvangent) [inline PokeAlarm/WebhookStructs.py]: Instead of 0, default to 'unkn' so that we can tell when it is missing vs just unset.

**Comment by myntath** [comment]
I think that last commit with the logo might be a mistake.

**Comment by Thashiznets** [comment]
@myntath yea git n00b sorry

**Comment by ifindpokemons** [comment]
@myntath yea soz git n00b meant to put it in my personal Image-Branch

**Comment by imp3tus** [comment]
any updates to this PR? Conflicts in it now.

**Comment by kvangent** [comment]
Added this in a301149b31969673265eb40710a195c192665987. Feel free to review or test it.

**Comment by ifindpokemons** [comment]
Sweet as, glad it's in, cheers Deadly.

**Pull request closed without merging**